backend/pkg/api/data_access/vdb_slotviz.go

```diff
@@ -35,6 +34,23 @@ func (d *DataAccessService) GetValidatorDashboardSlotViz(ctx context.Context, da
 		return nil, err
 	}
 
+	latestProposedSlot := int64(-1)
+	for slot := dutiesInfo.LatestSlot; ; slot-- {
+		if _, ok := dutiesInfo.PropAssignmentsForSlot[slot]; ok {
+			if dutiesInfo.SlotStatus[slot] == 1 {
+				latestProposedSlot = int64(slot)
+				break
+			}
+		} else {
+			// No more data available
+			break
+		}
+
+		if slot == 0 {
+			break
+		}
+	}
+
```

Comment:
thoughts about moving this to the slot viz service so we dont have to replicate this when something else needs the same logic? maybe a new `LatestProposedSlot` field in the `SyncData` struct?

Reply:
Sure, it's just one extra value in the cache.